internal/provider/helpers.go

```diff
@@ -126,6 +138,42 @@ func (*suppressAutoscaledDiskDiff) PlanModifyInt64(ctx context.Context, req plan
 	}
 }
 
+type clickhouseCustomCertificateValidator struct{}
+
+var _ validator.Object = &clickhouseCustomCertificateValidator{}
```

Comment:
Instead of custom validator, builtin one could be used

`func clickhouseCustomCertificateSchema() map[string]schema.Attribute {
	return map[string]schema.Attribute{
		"certificate": schema.StringAttribute{
			Optional:            true,
			MarkdownDescription: "Public certificate",
			PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
			Validators: []validator.String{
				stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("key")),
				stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("root_ca")),
			},
		},
		"key": schema.StringAttribute{
			Optional:            true,
			MarkdownDescription: "Private certificate key",
			PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
			Validators: []validator.String{
				stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("certificate")),
				stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("root_ca")),
			},
		},
		"root_ca": schema.StringAttribute{
			Optional:            true,
			MarkdownDescription: "Root certificate",
			PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
			Validators: []validator.String{
				stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("key")),
				stringvalidator.AlsoRequires(path.MatchRelative().AtParent().AtName("certificate")),
			},
		},
	}
}`